### DISCUSSION TITLE
Status 3: How to know what Setoption are active?

### DISCUSSION THREAD

**Discussion by ozett** [arendst/Tasmota#17992] [Q&A] [answered]
i want to see my active setoptions
i got output of status 0 and status3 and see "encrypted" setoptions in the log

setoptions only:
```
STATUS3 = {"SetOption":["00000009","2805C80001000600003C5A0A192800000000","40000080","00006000","00004000","00000000"]}}
```
full output:
```
STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Seamans","Seamans-ext"],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00000009","2805C80001000600003C5A0A192800000000","40000080","00006000","00004000","00000000"]}}
```
but i cannot read this to reproduce setoptions x on another device.

how are this loginformation converted to "normal" setoption?

**Comment by barbudor** [comment, marked as answer]
You can use tools such as decode-status or decode-config
https://github.com/arendst/Tasmota/blob/development/tools/decode-status.py
https://github.com/tasmota/decode-config

**Reply by sfromis** [reply]
If you prefer to parse the array from `Status 3`, each of the 8-digit blocks is 32 options with the first block being option 31..0, and so on counting up, while the longer block is 8-bit numbers for option 32..49. But probably simpler just using decode-status :nerd_face:

**Reply by ozett** [reply]
That's an answer that's really fun. intelligent and humorous. And helpful. Maybe I'll try to decode the blocks... Please continue like this, so that the forum doesn't always only know the typical dry answers. (They also help of course) 🤓 thanks

**Comment by fragolinux** [comment]
anyone wants to give a try to this HACS integration i just created, based on this decoder script? I tried in 2 different installs and works good... it's perfect? Doubt, but does its job... it's a 1 hour attempt to create a HACS plugin using chatgpt for help... i never programmed in python nor know more than this, but eager to learn :)

https://github.com/fragolinux/tasmota_decoder

![image](https://github.com/arendst/Tasmota/assets/21192010/dc14a17d-edfd-4ca3-9512-d7b11a101d74)
